### DISCUSSION TITLE
Dynamic Routing in Nuxt

### DISCUSSION THREAD

**Discussion by krishna15873** [nuxt/nuxt#10694] [Questions (Nuxt 2)]
Hi All,

We have a CMS that we're trying to migrate to Nuxt. We have dynamic URls like this:

/
/news
/news/url-of-article
/blog
/any-other-page

How do we configure dynamic routes like this in Nuxt? Any help would be highly appreciated.

Thanks!!

**Comment by ghost** [comment]
That depends a bit on the type of CMS you are using. At the ecompany I work at, we use nuxt with Storyblok. And most of the time the pages-directory we have just 2 files: index.vue and _.vue .
A _.vue file creates a catch-all-route. If the current path isn't a corresponding story on Storyblok, a 404-error is thrown.

Alternatively, there's the option of dynamic routes, where you can also name the route-param and retrieve it from the Context and go from there: https://nuxtjs.org/docs/features/file-system-routing#dynamic-routes

**Reply by ghost** [reply]
This also works for static generation: The Nuxt build will crawl through the possible routes automatically.

**Reply by ghost** [reply]
In Nuxt3 the principal remains the same, but the file-naming is a bit different. Route params are inside array-brackets in the file's name and the catch-all routes are generated with a page-file called '[...slug].vue': https://v3.nuxtjs.org/guide/directory-structure/pages#catch-all-route
